### ISSUE TITLE
Dependency Tracking for .NET Core

### ISSUE THREAD

**Issue by billpratt** [Microsoft/ApplicationInsights-aspnetcore#298]
I am currently adding application insights to an existing ASP.NET Core web app running on .NET Core. While testing, I noticed remote HTTP calls (dependency tracking) are not being collected. Reading this [wiki article](https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Dependency-Tracking-and-Performance-Counter-Collection) it stated that `Currently not supported in .NET Core.`

Is support for this on the roadmap?

**Comment by billpratt** [comment]
![image](https://cloud.githubusercontent.com/assets/2847527/21187065/a2bdb20e-c1e4-11e6-90e2-8e06d2428f61.png)

This is what I'm currently seeing

**Comment by kartang** [comment]
Hello,
Sorry for the inconvenience. 
That's correct! Dependency tracking is only supported in full framework as of today. We have some plans of supporting it in core framework, but nothing is on the roadmap yet/finalized.
Thank you

**Comment by billpratt** [comment]
That's disappointing but understandable. I really wanted dependency tracking for my asp.net core app.

**Comment by Peter-Juhasz** [comment]
Yes, we use and **love** Application Insights, but we were really disappointed when we found out that there is no dependency tracking at all. Now it feels like being blind at some of our new services, which we've built using ASP.NET Core already. And also we are unable to access the full set of features the service provides. Please prioritize it high in the backlog, because dependency tracking is a very powerful feature and we need it.

If it needs infrastructure support from .NET Core, please improve on the collaboration between the teams inside the company, because it makes introducing new technologies very hard when they lack key features.

**Comment by Peter-Juhasz** [comment]
Awesome, Entity Framework is our number one dependency source, while Azure Storage (or more generally HTTP) is the second. Can't wait to see them in the Azure portal.

**Comment by billpratt** [comment]
Great to hear. What about tracking for external service calls? Would be a great addition to ASP.NET Core used as microservices

**Comment by billpratt** [comment]
In the latest ASP.NET community stand up @DamianEdwards stated that they are working on enabling HTTP dependency tracking in ASP.NET Core. Any ideas on timeframe? Currently I'm using the TelemetryClient to track HTTP dependencies and works fine.  Having an idea of when the feature is coming would be nice but I understand if it's too early to know.

**Comment by DamianEdwards** [comment]
@billpratt you can do it manually today in ASP.NET Core of course but indeed we're working to make this automatic, much like it is in System.Web-based ASP.NET. Super-rough timeframe is in the next few months but we're trying to figure out how to make it happen ASAP. Will keep this issue updated on the progress.

**Comment by billpratt** [comment]
Thanks @DamianEdwards. I'm excited to see traction on this and what is to come!

**Comment by billpratt** [comment]
Just checking in to see if there is any update on this? @DamianEdwards

**Comment by DamianEdwards** [comment]
It's moved slightly forward but I believe we're waiting on some build infrastructure to be updated for VS2017 RTM before we can take the next steps. Possible we'll have some preview of this in the next month, but I can't say for sure.

**Comment by billpratt** [comment]
Any way the community can get involved?

**Comment by oscarwest** [comment]
How is this coming along?
Are the following features planned?
Entity Framework
Http Requests
WCF

**Comment by johnkwaters** [comment]
Any news ? My stack uses ED core and Azure Db and I would love to include those dependencies

**Comment by dnduffy** [comment]
The release is planned for this week (or worst case scenario next Monday).
At this point we are just validating the build before releasing it.

**Comment by htuomola** [comment]
Sorry if this is not the best place but is there a summary anywhere on what is/isn't supported? I have an ASP.net Core web application which runs on full .NET framework (4.6.1). The site is hosted on Azure App Services. The code is calling Azure DocumentDB, Azure Redis, PostgreSQL (via Entity Framework 6) and Elasticsearch and out of these, only the Elasticsearch HTTP calls are logged as dependencies. Is that to be expected or do I have some issues with config? If yes, what could it be?

**Comment by dnduffy** [comment]
@htuomola Please file a new separate issue.  Thanks.
